### PR TITLE
[DO NOT MERGE] Release 118 - Bad rows

### PR DESCRIPTION
**Pull request by BenFradet**
- [x] one jsonpath lib, maybe json-surfer because of https://github.com/circe/circe-jackson/issues/49 (impacted are pii, sql and api)
- [x] no ttl for scala-lru-map in the sql query enrichment
- [x] diffing for pii?
- [x] make `better-monadic-for` work
- [x] api request `Tags` => specific semigroup
- [x] referer parser `Lens`
- [x] referer parser 2.0
- [ ] update referer parser enrichment config example (related to #4003 )
- [x] travis 2.12
- [ ] rm tmp iglu
- [x] rm bintray resolver
- [ ] clean-up #4122 and #4238 during rebase (was added in #4218)
- [ ] rename the branch

### CODE REVIEW

File: 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/RawEvent.scala [hunk 1]
```diff
@@ -43,7 +42,7 @@ object RawEvent {
       re.source.name,
       re.source.encoding,
       re.source.hostname,
-      re.context.timestamp.map(_.toString),
+      re.context.timestamp,
```
Comment (benjben) [inline 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/RawEvent.scala]: Love this! :+1:

File: 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/registry/Adapter.scala [hunk 2]
```diff
@@ -174,26 +181,17 @@ trait Adapter {
   protected[registry] def toUnstructEventParams(
     tracker: String,
     parameters: RawEventParameters,
-    schema: String,
+    schema: SchemaKey,
```
Comment (benjben) [inline 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/adapters/registry/Adapter.scala]: Awesome!

File: 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/SchemaEnrichment.scala [hunk 3]
```diff
@@ -89,4 +93,13 @@ object SchemaEnrichment {
       case _ => Monad[F].pure(possibleFailure)
     }
   }
+
+  /** Parse string, which is certainly JSON */
+  private def unsafeParseJson(string: String): Json =
```
[the remark this answers is not in the document]
Comment (chuwy) [reply 3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/SchemaEnrichment.scala]: Addressed this in https://github.com/snowplow/snowplow/issues/4233

**Comment by benjben** [comment]
Handled by #4254

**Pull request closed without merging**